src/tests/krate.rs

```diff
@@ -97,173 +96,136 @@ fn index() {
 
 #[test]
 fn index_queries() {
-    let (_b, app, middle) = app();
-
-    let u;
-    let krate;
-    let krate2;
-    {
-        let conn = app.diesel_database.get().unwrap();
-        u = new_user("foo").create_or_update(&conn).unwrap();
+    let (app, anon, user) = TestApp::with_user();
+    let user = user.as_model();
 
-        krate = CrateBuilder::new("foo_index_queries", u.id)
+    let (krate, krate2) = app.db(|conn| {
+        let krate = CrateBuilder::new("foo_index_queries", user.id)
             .readme("readme")
             .description("description")
             .keyword("kw1")
-            .expect_build(&conn);
+            .expect_build(conn);
 
-        krate2 = CrateBuilder::new("BAR_INDEX_QUERIES", u.id)
+        let krate2 = CrateBuilder::new("BAR_INDEX_QUERIES", user.id)
             .keyword("KW1")
-            .expect_build(&conn);
+            .expect_build(conn);
 
-        CrateBuilder::new("foo", u.id)
+        CrateBuilder::new("foo", user.id)
             .keyword("kw3")
-            .expect_build(&conn);
-    }
+            .expect_build(conn);
+        (krate, krate2)
+    });
 
-    let mut req = req(Method::Get, "/api/v1/crates");
-    let mut response = ok_resp!(middle.call(req.with_query("q=baz")));
-    assert_eq!(::json::<CrateList>(&mut response).meta.total, 0);
+    assert_eq!(anon.search("q=baz").meta.total, 0);
 
     // All of these fields should be indexed/searched by the queries
-    let mut response = ok_resp!(middle.call(req.with_query("q=foo")));
-    assert_eq!(::json::<CrateList>(&mut response).meta.total, 2);
-    let mut response = ok_resp!(middle.call(req.with_query("q=kw1")));
-    assert_eq!(::json::<CrateList>(&mut response).meta.total, 2);
-    let mut response = ok_resp!(middle.call(req.with_query("q=readme")));
-    assert_eq!(::json::<CrateList>(&mut response).meta.total, 1);
-    let mut response = ok_resp!(middle.call(req.with_query("q=description")));
-    assert_eq!(::json::<CrateList>(&mut response).meta.total, 1);
-
-    let query = format!("user_id={}", u.id);
-    let mut response = ok_resp!(middle.call(req.with_query(&query)));
-    assert_eq!(::json::<CrateList>(&mut response).crates.len(), 3);
-    let mut response = ok_resp!(middle.call(req.with_query("user_id=0")));
-    assert_eq!(::json::<CrateList>(&mut response).crates.len(), 0);
+    assert_eq!(anon.search("q=foo").meta.total, 2);
+    assert_eq!(anon.search("q=kw1").meta.total, 2);
+    assert_eq!(anon.search("q=readme").meta.total, 1);
+    assert_eq!(anon.search("q=description").meta.total, 1);
```

Comment:
ugh this is so much nicer i want to cry tears of joy! i can actually see what's being tested!! 😻😻😻